### DISCUSSION TITLE
when i use ? in next.config.js for redirect , i receive the error

### DISCUSSION THREAD

**Discussion by 1377sayaba** [vercel/next.js#40783] [Help]
when i want redirect the link below
source: 'product/offers?menu=sheetSelect&value=ورق%20خام'
the error is 
![image](https://user-images.githubusercontent.com/89265632/191696140-d942a005-3a79-43e2-acbc-2a21fa235b19.png)

is it have a way for redirect this link?

**Comment by icyJoseph** [comment]
Redirect `paths` in `next.config.js` cannot contain queries, because the `?` is a reserver Regexp character. 

> The following characters (, ), {, }, :, *, +, ? are used for regex path matching, so when used in the source as non-special values they must be escaped by adding \\ before them

Instead use query matching like https://nextjs.org/docs/api-reference/next.config.js/redirects#header-cookie-and-query-matching describes:

```js
    {
        source: '/specific/:path*',
        has: [
          {
            type: 'query',
            key: 'page',
            // the page value will not be available in the
            // destination since value is provided and doesn't
            // use a named capture group e.g. (?<page>home)
            value: 'home',
          },
        ],
```

I have some trouble getting the actual string you use, but for example this should work:

```js
/** @type {import('next').NextConfig} */
const nextConfig = {
  reactStrictMode: true,
  async redirects() {
    return [
      {
        source: "/product/offers",
        destination: "/",
        has: [
          {
            type: "query",
            key: "menu",
            value: "sheetSelect",
          },
          {
            type: "query",
            key: "value",
            value: "thevalue",
          },
        ],
        permanent: false,
      },
    ];
  },
};

module.exports = nextConfig;
```

Then going to `http://localhost:3000/product/offers?menu=sheetSelect&value=thevalue` does the redirect to `/`.
